blaze/compute/sql.py

```diff
@@ -789,3 +792,37 @@ def post_compute(_, s, **kwargs):
 @dispatch(IsIn, ColumnElement)
 def compute_up(expr, data, **kwargs):
     return data.in_(expr._keys)
+
+
+@dispatch(Slice, ClauseElement)
```

Comment:
`ClauseElement` is too general here. It is the base class for *every* SQLAlchemy expression. We only want to slice the following things: `ColumnElement`s, `Selectable`s, and `Select` statements. `ColumnElement` includes things like `Column`s and any function on `Column`s. A `Selectable` is anything that can show up in the `from` clause in a `select` statement.

I'd dispatch like this:

```python
@dispatch(Slice, (ColumnElement, Selectable, Select))
```